### PR TITLE
Error handling

### LINKED ISSUE #46: Report error when you can't register a user.
If I try to create a user without an `email` or `password`, the server doesn't let me.  But I should see the reason why.

cursor_and_localhost_5000_dev_html__user

The same thing should happen if you try to update a user's password.

### LINKED ISSUE #51: Players must have a name
Prevent a player from being created if he/she doesn't have a name.

cursor_and_localhost_5000_dev_html__players

This should be done by making it a required field in the DB by creating a migration that does it.  Then the error from bookshelf should be sent back to the client.  Hopefully the client can put it near the create button.

Also, if you try to update a player and remove the name, it should not work.

### LINKED ISSUE #53: Tournaments must have a date
Similar to #51, we should make sure tournaments must have a date on the DB level.  Then we should propagate that error to the client and show it next to the create button.

### LINKED ISSUE #71: Handle model/server errors
For #46, #48, #51, and #53, we need to propagate server errors to display them on the client.

Since we're in the "make it work" phase and not the "make it pretty" phase, this should just include an error message by the submit button on the relevant forms.

### PR DESCRIPTION
**Pull request by jandjorgensen**
Fixes #71 
Fixes #53 
Fixes #51 
Fixes #46 

Blocked on #70 since this relies on some of the fixture & test updates from that PR.

Migrations are not really "the right place" to handle required fields (the only example of this type of thing in the bookshelf or knex documentation occurs in the model).

### CODE REVIEW

**Comment by justinbmeyer** [comment]
Having this sort of protection in the database is absolutely the right way of doing it. Regardless if bookshelf can do it in the model layer as well. 

Sometimes other apps might write to a db. Or an admin needs to make changes directly to the DB.  Having some scheme rules there prevents invalid data for all.

**Comment by jandjorgensen** [comment]
I think this is the way I actually should have done the validation step:

```
// in a bookshelf model
intialize: function(){
    this.on('saving', this.validateSave)
},
validateSave: function(){
   // validate
   return isValid;
}
```

File: migrations/20150809023413-add-stats-time.js [hunk 1]
```diff
@@ -12,6 +12,6 @@ exports.up = function(db, callback) {
 exports.down = function(db, callback) {
 	async.series([
 		db.removeColumn.bind(db, "stats","time"),
-		db.addCremoveColumnolumn.bind(db, "stats","value")
+		db.removeColumn.bind(db, "stats","value")
```
Comment (justinbmeyer) [inline migrations/20150809023413-add-stats-time.js]: I thought I fixed this.  Maybe master needs to be pulled into staging.

File: public/components/player/edit/edit.stache [hunk 2]
```diff
@@ -37,4 +37,9 @@
   </div>
   <button type="submit" class="btn btn-primary" >{{#player.isNew}}Create{{else}}Update{{/player.isNew}}</button>
   <a class="btn btn-default" ($click)="cancelEvent(%element)">Cancel</a>
+  {{#is savePromise.state 'rejected'}}
```
Comment (justinbmeyer) [inline public/components/player/edit/edit.stache]: @jandjorgensen You should be able to do this like `{{#savePromise.isRejected}}`

File: public/components/player/edit/edit.stache [hunk 3]
```diff
@@ -37,4 +37,9 @@
...
+    {{#each savePromise.reason.responseJSON}}
+      <p>{{.}}</p>
```
Comment (justinbmeyer) [inline public/components/player/edit/edit.stache]: can these `<p>` have a `class="error"` on them that are styled red?

File: public/components/player/edit/edit_test.js [hunk 4]
```diff
@@ -57,10 +55,30 @@ QUnit.module('player/edit', function(hooks){
...
+					"weight": 200,
+					"height": 71,
+					"birthday": "1980-01-01"
+				},
+				playerModel = new Player(player),
+				vam = new ViewModel({
+					player: playerModel
+				});
+
+				vm.savePlayer().fail(function(resp, type){
```
Comment (justinbmeyer) [inline public/components/player/edit/edit_test.js]: we shouldn't rely on `savePlayer` returning a promise b/c the template wouldn't.  Instead we should probably call:

```
vm.savePlayer();
vm.attr("savePromise").then(function(){...}, function(){...})
```

File: services/players.js [hunk 5]
```diff
@@ -44,11 +44,12 @@ app['delete']('/services/players/:id', function(req, res){
...
-		res.send({id: player.get('id')});
-	}, function(e){
-		res.status(500).send(e);
-	});
+		new Player(clean(req.body)).save().then(function(player){
+			res.send({id: player.get('id')});
+		}, function(e){
+			res.status(500).send(e);
+		});
+	// }
```
Comment (justinbmeyer) [inline services/players.js]: other than incorrect indentation, it's hard to tell what the difference is?

**Pull request merged**